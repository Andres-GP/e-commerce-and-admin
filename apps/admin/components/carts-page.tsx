"use client";

import { useEffect, useState } from "react";
import { Card } from "@/components/ui/card";
import { ShoppingCart, Package } from "lucide-react";
import { cn } from "@/lib/utils";

interface CartItem {
  _id: string;
  userId: string;
  productId: string;
  name: string;
  price: number;
  quantity: number;
  image?: string;
  addedAt: string;
}

export function CartsPage() {
  const [carts, setCarts] = useState<CartItem[]>([]);
  const [loading, setLoading] = useState(true);

  console.log(carts, "carts");
  useEffect(() => {
    fetchCarts();
  }, []);

  const fetchCarts = async () => {
    try {
      const res = await fetch("/api/cart");
      const data = await res.json();
      console.log(data, "data");
      setCarts(data.data || []);
    } catch (error) {
      console.error("Failed to fetch carts:", error);
    } finally {
      setLoading(false);
    }
  };

  const groupedCarts = carts.reduce(
    (acc, item) => {
      if (!acc[item.userId]) {
        acc[item.userId] = [];
      }
      acc[item.userId].push(item);
      return acc;
    },
    {} as Record<string, CartItem[]>
  );

  return (
    <div className="p-8 space-y-6 animate-fade-in-up">
      <div>
        <h2 className="text-3xl font-bold text-balance">Active Carts</h2>
        <p className="text-muted-foreground mt-1">
          Monitor customer shopping carts
        </p>
      </div>

      {loading ? (
        <div className="space-y-4">
          {[...Array(3)].map((_, i) => (
            <Card key={i} className="p-6">
              <div className="h-24 bg-muted animate-pulse rounded-lg" />
            </Card>
          ))}
        </div>
      ) : Object.keys(groupedCarts).length === 0 ? (
        <Card className="p-12">
          <div className="text-center">
            <ShoppingCart className="w-12 h-12 text-muted-foreground mx-auto mb-4" />
            <p className="text-muted-foreground">No active carts</p>
          </div>
        </Card>
      ) : (
        <div className="space-y-4">
          {Object.entries(groupedCarts).map(([userId, items], index) => {
            const total = items.reduce(
              (sum, item) => sum + item.price * item.quantity,
              0
            );

            return (
              <Card
                key={userId}
                className={cn(
                  "p-6 transition-all duration-300 hover:shadow-lg hover:scale-[1.01]",
                  "animate-fade-in-up"
                )}
                style={{ animationDelay: `${index * 50}ms` }}
              >
                <div className="flex items-center justify-between mb-4">
                  <div>
                    <h4 className="font-semibold">User: {userId}</h4>
                    <p className="text-sm text-muted-foreground">
                      {items.length} items in cart
                    </p>
                  </div>
                  <div className="text-right">
                    <p className="text-2xl font-bold text-primary">
                      ${total.toFixed(2)}
                    </p>
                    <p className="text-sm text-muted-foreground">Cart Total</p>
                  </div>
                </div>

                <div className="space-y-2">
                  {items.map((item) => (
                    <div
                      key={item._id}
                      className="flex items-center gap-3 p-3 rounded-lg bg-accent/50"
                    >
                      <div className="w-12 h-12 bg-muted rounded flex items-center justify-center">
                        <Package className="w-6 h-6 text-muted-foreground" />
                      </div>

                      <div className="flex-1">
                        <p className="font-medium text-sm">{item.name}</p>
                        <p className="text-xs text-muted-foreground">
                          ${item.price.toFixed(2)} Ã— {item.quantity}
                        </p>
                      </div>

                      <p className="font-semibold">
                        ${(item.price * item.quantity).toFixed(2)}
                      </p>
                    </div>
                  ))}
                </div>
              </Card>
            );
          })}
        </div>
      )}
    </div>
  );
}
