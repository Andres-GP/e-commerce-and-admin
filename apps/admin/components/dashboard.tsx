"use client";

import { useEffect, useState } from "react";
import { Card } from "@/components/ui/card";
import {
  Package,
  ShoppingBag,
  Star,
  TrendingUp,
  DollarSign,
  Users,
} from "lucide-react";
import { cn } from "@/lib/utils";

interface DashboardProps {
  onNavigate: (
    page: "dashboard" | "products" | "orders" | "reviews" | "carts"
  ) => void;
}

export function Dashboard({ onNavigate }: DashboardProps) {
  const [stats, setStats] = useState({
    products: 0,
    orders: 0,
    reviews: 0,
    revenue: 0,
  });
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchStats = async () => {
      try {
        const [productsRes, ordersRes, reviewsRes] = await Promise.all([
          fetch("/api/products"),
          fetch("/api/orders"),
          fetch("/api/reviews"),
        ]);

        const [products, orders, reviews] = await Promise.all([
          productsRes.json(),
          ordersRes.json(),
          reviewsRes.json(),
        ]);

        const revenue =
          orders.data?.reduce(
            (sum: number, order: any) => sum + (order.total || 0),
            0
          ) || 0;

        setStats({
          products: products.total || 0,
          orders: orders.total || 0,
          reviews: reviews.total || 0,
          revenue,
        });
      } catch (error) {
        console.error("Failed to fetch stats:", error);
      } finally {
        setLoading(false);
      }
    };

    fetchStats();
  }, []);

  const statCards = [
    {
      title: "Total Products",
      value: stats.products,
      icon: Package,
      color: "from-chart-1 to-chart-2",
      page: "products" as const,
    },
    {
      title: "Total Orders",
      value: stats.orders,
      icon: ShoppingBag,
      color: "from-chart-2 to-chart-3",
      page: "orders" as const,
    },
    {
      title: "Total Reviews",
      value: stats.reviews,
      icon: Star,
      color: "from-chart-4 to-chart-5",
      page: "reviews" as const,
    },
    {
      title: "Total Revenue",
      value: `$${stats.revenue.toFixed(2)}`,
      icon: DollarSign,
      color: "from-chart-5 to-chart-1",
      page: "orders" as const,
    },
  ];

  return (
    <div className="p-8 space-y-8 animate-fade-in-up">
      <div>
        <h2 className="text-3xl font-bold text-balance">Welcome back, Admin</h2>
        <p className="text-muted-foreground mt-2">
          Here's what's happening with your store today.
        </p>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
        {statCards.map((stat, index) => {
          const Icon = stat.icon;
          return (
            <Card
              key={stat.title}
              onClick={() => onNavigate(stat.page)}
              className={cn(
                "p-6 cursor-pointer transition-all duration-300 hover:scale-105 hover:shadow-xl",
                "animate-fade-in-up border-border/50 backdrop-blur-sm",
                "group relative overflow-hidden"
              )}
              style={{ animationDelay: `${index * 100}ms` }}
            >
              <div
                className={cn(
                  "absolute inset-0 bg-gradient-to-br opacity-5 group-hover:opacity-10 transition-opacity",
                  stat.color
                )}
              />

              <div className="relative">
                <div className="flex items-center justify-between mb-4">
                  <div
                    className={cn(
                      "p-3 rounded-lg bg-gradient-to-br",
                      stat.color,
                      "shadow-lg"
                    )}
                  >
                    <Icon className="w-6 h-6 text-white" />
                  </div>
                  <TrendingUp className="w-5 h-5 text-chart-3 opacity-0 group-hover:opacity-100 transition-opacity" />
                </div>

                <p className="text-sm text-muted-foreground mb-1">
                  {stat.title}
                </p>
                <p className="text-3xl font-bold">
                  {loading ? (
                    <span className="inline-block w-20 h-8 bg-muted animate-pulse rounded" />
                  ) : (
                    stat.value
                  )}
                </p>
              </div>
            </Card>
          );
        })}
      </div>

      <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
        <Card
          className="p-6 animate-fade-in-up"
          style={{ animationDelay: "400ms" }}
        >
          <h3 className="text-xl font-semibold mb-4 flex items-center gap-2">
            <TrendingUp className="w-5 h-5 text-primary" />
            Quick Actions
          </h3>
          <div className="space-y-3">
            <button
              onClick={() => onNavigate("products")}
              className="w-full p-4 text-left rounded-lg bg-accent hover:bg-accent/80 transition-all duration-200 hover:scale-[1.02] active:scale-[0.98] group"
            >
              <p className="font-medium group-hover:text-primary transition-colors">
                Manage Products
              </p>
              <p className="text-sm text-muted-foreground text-white">
                Add, edit, or remove products
              </p>
            </button>
            <button
              onClick={() => onNavigate("orders")}
              className="w-full p-4 text-left rounded-lg bg-accent hover:bg-accent/80 transition-all duration-200 hover:scale-[1.02] active:scale-[0.98] group"
            >
              <p className="font-medium group-hover:text-primary transition-colors">
                View Orders
              </p>
              <p className="text-sm text-muted-foreground text-white">
                Check order status and details
              </p>
            </button>
            <button
              onClick={() => onNavigate("reviews")}
              className="w-full p-4 text-left rounded-lg bg-accent hover:bg-accent/80 transition-all duration-200 hover:scale-[1.02] active:scale-[0.98] group"
            >
              <p className="font-medium group-hover:text-primary transition-colors">
                Moderate Reviews
              </p>
              <p className="text-sm text-muted-foreground text-white">
                Manage customer feedback
              </p>
            </button>
          </div>
        </Card>

        <Card
          className="p-6 animate-fade-in-up"
          style={{ animationDelay: "500ms" }}
        >
          <h3 className="text-xl font-semibold mb-4 flex items-center gap-2">
            <Users className="w-5 h-5 text-primary" />
            Recent Activity
          </h3>
          <div className="space-y-4">
            <div className="flex items-start gap-3 p-3 rounded-lg bg-accent/50">
              <div className="w-2 h-2 rounded-full bg-chart-3 mt-2" />
              <div>
                <p className="text-sm font-medium">New order received</p>
                <p className="text-xs text-muted-foreground">2 minutes ago</p>
              </div>
            </div>
            <div className="flex items-start gap-3 p-3 rounded-lg bg-accent/50">
              <div className="w-2 h-2 rounded-full bg-chart-1 mt-2" />
              <div>
                <p className="text-sm font-medium">Product updated</p>
                <p className="text-xs text-muted-foreground">15 minutes ago</p>
              </div>
            </div>
            <div className="flex items-start gap-3 p-3 rounded-lg bg-accent/50">
              <div className="w-2 h-2 rounded-full bg-chart-5 mt-2" />
              <div>
                <p className="text-sm font-medium">New review posted</p>
                <p className="text-xs text-muted-foreground">1 hour ago</p>
              </div>
            </div>
          </div>
        </Card>
      </div>
    </div>
  );
}
