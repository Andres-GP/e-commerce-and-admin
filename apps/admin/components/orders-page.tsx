"use client";

import { useEffect, useState } from "react";
import { Card } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Search, ShoppingBag, MapPin, CreditCard, Package } from "lucide-react";
import { cn } from "@/lib/utils";

interface Order {
  _id: string;
  userId: string;
  items: Array<{
    name: string;
    quantity: number;
    price: number;
    image?: string;
  }>;
  shippingAddress: {
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    city: string;
    state: string;
    zip: string;
  };
  paymentMethod: string;
  total: number;
  status: string;
  createdAt: string;
}

export function OrdersPage() {
  const [orders, setOrders] = useState<Order[]>([]);
  const [loading, setLoading] = useState(true);
  const [searchTerm, setSearchTerm] = useState("");
  const [expandedOrder, setExpandedOrder] = useState<string | null>(null);

  useEffect(() => {
    fetchOrders();
  }, []);

  const fetchOrders = async () => {
    try {
      const res = await fetch("/api/orders");
      const data = await res.json();
      setOrders(data.data || []);
    } catch (error) {
      console.error("Failed to fetch orders:", error);
    } finally {
      setLoading(false);
    }
  };

  const filteredOrders = orders.filter(
    (order) =>
      order.shippingAddress.email
        .toLowerCase()
        .includes(searchTerm.toLowerCase()) ||
      order._id.toLowerCase().includes(searchTerm.toLowerCase())
  );

  return (
    <div className="p-8 space-y-6 animate-fade-in-up">
      <div>
        <h2 className="text-3xl font-bold text-balance">Orders</h2>
        <p className="text-muted-foreground mt-1">
          View and manage customer orders
        </p>
      </div>

      <Card className="p-6">
        <div className="flex items-center gap-4 mb-6">
          <div className="relative flex-1">
            <Search className="absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground" />
            <Input
              placeholder="Search by email or order ID..."
              value={searchTerm}
              onChange={(e) => setSearchTerm(e.target.value)}
              className="pl-10"
            />
          </div>
        </div>

        {loading ? (
          <div className="space-y-4">
            {[...Array(3)].map((_, i) => (
              <div key={i} className="h-32 bg-muted animate-pulse rounded-lg" />
            ))}
          </div>
        ) : filteredOrders.length === 0 ? (
          <div className="text-center py-12">
            <ShoppingBag className="w-12 h-12 text-muted-foreground mx-auto mb-4" />
            <p className="text-muted-foreground">No orders found</p>
          </div>
        ) : (
          <div className="space-y-4">
            {filteredOrders.map((order, index) => (
              <Card
                key={order._id}
                className={cn(
                  "p-6 cursor-pointer transition-all duration-300",
                  "hover:shadow-lg hover:scale-[1.01] animate-fade-in-up",
                  expandedOrder === order._id && "ring-2 ring-primary"
                )}
                style={{ animationDelay: `${index * 50}ms` }}
                onClick={() =>
                  setExpandedOrder(
                    expandedOrder === order._id ? null : order._id
                  )
                }
              >
                <div className="flex items-start justify-between mb-4">
                  <div>
                    <div className="flex items-center gap-3 mb-2">
                      <h4 className="font-semibold">
                        Order #{order._id.slice(-8)}
                      </h4>
                      <span className="px-3 py-1 rounded-full text-xs font-medium bg-chart-1/20 text-chart-1">
                        {order.status}
                      </span>
                    </div>
                    <p className="text-sm text-muted-foreground">
                      {new Date(order.createdAt).toLocaleDateString("en-US", {
                        year: "numeric",
                        month: "long",
                        day: "numeric",
                      })}
                    </p>
                  </div>
                  <div className="text-right">
                    <p className="text-2xl font-bold text-primary">
                      ${order.total.toFixed(2)}
                    </p>
                    <p className="text-sm text-muted-foreground">
                      {order.items.length} items
                    </p>
                  </div>
                </div>

                {expandedOrder === order._id && (
                  <div className="space-y-4 pt-4 border-t border-border animate-scale-in">
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                      <div className="space-y-2">
                        <div className="flex items-center gap-2 text-sm font-medium">
                          <MapPin className="w-4 h-4 text-primary" />
                          Shipping Address
                        </div>
                        <div className="text-sm text-muted-foreground pl-6">
                          <p>
                            {order.shippingAddress.firstName}{" "}
                            {order.shippingAddress.lastName}
                          </p>
                          <p>{order.shippingAddress.email}</p>
                          <p>{order.shippingAddress.address}</p>
                          <p>
                            {order.shippingAddress.city},{" "}
                            {order.shippingAddress.state}{" "}
                            {order.shippingAddress.zip}
                          </p>
                        </div>
                      </div>

                      <div className="space-y-2">
                        <div className="flex items-center gap-2 text-sm font-medium">
                          <CreditCard className="w-4 h-4 text-primary" />
                          Payment Method
                        </div>
                        <p className="text-sm text-muted-foreground pl-6">
                          {order.paymentMethod}
                        </p>
                      </div>
                    </div>

                    <div className="space-y-2">
                      <div className="flex items-center gap-2 text-sm font-medium">
                        <Package className="w-4 h-4 text-primary" />
                        Order Items
                      </div>
                      <div className="space-y-2 pl-6">
                        {order.items.map((item, idx) => (
                          <div
                            key={idx}
                            className="flex items-center gap-3 p-3 rounded-lg bg-accent/50"
                          >
                            <div className="w-12 h-12 bg-muted rounded flex items-center justify-center">
                              <Package className="w-6 h-6 text-muted-foreground" />
                            </div>
                            <div className="flex-1">
                              <p className="font-medium text-sm">{item.name}</p>
                              <p className="text-xs text-muted-foreground">
                                Qty: {item.quantity}
                              </p>
                            </div>
                            <p className="font-semibold">
                              ${(item.price * item.quantity).toFixed(2)}
                            </p>
                          </div>
                        ))}
                      </div>
                    </div>
                  </div>
                )}
              </Card>
            ))}
          </div>
        )}
      </Card>
    </div>
  );
}
