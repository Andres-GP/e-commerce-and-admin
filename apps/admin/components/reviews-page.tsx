"use client";

import { useEffect, useState } from "react";
import { Card } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Search, Star, Trash2 } from "lucide-react";
import { Button } from "@/components/ui/button";
import { cn } from "@/lib/utils";

interface Review {
  _id: string;
  productId?: number;
  name: string;
  rating: number;
  date: string;
  comment: string;
  avatar: string;
}

export function ReviewsPage() {
  const [reviews, setReviews] = useState<Review[]>([]);
  const [loading, setLoading] = useState(true);
  const [searchTerm, setSearchTerm] = useState("");

  useEffect(() => {
    fetchReviews();
  }, []);

  const fetchReviews = async () => {
    try {
      const res = await fetch("/api/reviews");
      const data = await res.json();
      setReviews(data.data || []);
    } catch (error) {
      console.error("Failed to fetch reviews:", error);
    } finally {
      setLoading(false);
    }
  };

  const filteredReviews = reviews.filter(
    (review) =>
      review.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
      review.comment.toLowerCase().includes(searchTerm.toLowerCase())
  );

  return (
    <div className="p-8 space-y-6 animate-fade-in-up">
      <div>
        <h2 className="text-3xl font-bold text-balance">Reviews</h2>
        <p className="text-muted-foreground mt-1">
          Manage customer reviews and feedback
        </p>
      </div>

      <Card className="p-6">
        <div className="flex items-center gap-4 mb-6">
          <div className="relative flex-1">
            <Search className="absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground" />
            <Input
              placeholder="Search reviews..."
              value={searchTerm}
              onChange={(e) => setSearchTerm(e.target.value)}
              className="pl-10"
            />
          </div>
        </div>

        {loading ? (
          <div className="space-y-4">
            {[...Array(3)].map((_, i) => (
              <div key={i} className="h-32 bg-muted animate-pulse rounded-lg" />
            ))}
          </div>
        ) : filteredReviews.length === 0 ? (
          <div className="text-center py-12">
            <Star className="w-12 h-12 text-muted-foreground mx-auto mb-4" />
            <p className="text-muted-foreground">No reviews found</p>
          </div>
        ) : (
          <div className="space-y-4">
            {filteredReviews.map((review, index) => (
              <Card
                key={review._id}
                className={cn(
                  "p-6 transition-all duration-300 hover:shadow-lg hover:scale-[1.01]",
                  "animate-fade-in-up"
                )}
                style={{ animationDelay: `${index * 50}ms` }}
              >
                <div className="flex items-start gap-4">
                  <div className="flex-1">
                    <div className="flex items-center justify-between mb-2">
                      <div>
                        <h4 className="font-semibold">{review.name}</h4>
                        <p className="text-sm text-muted-foreground">
                          {review.date}
                        </p>
                      </div>

                      <div className="flex items-center gap-2">
                        <div className="flex items-center gap-1">
                          {[...Array(5)].map((_, i) => (
                            <Star
                              key={i}
                              className={cn(
                                "w-4 h-4",
                                i < review.rating
                                  ? "fill-chart-5 text-chart-5"
                                  : "text-muted-foreground"
                              )}
                            />
                          ))}
                        </div>

                        <Button
                          disabled
                          variant="ghost"
                          size="icon"
                          className="hover:bg-destructive/10 hover:text-destructive transition-colors"
                        >
                          <Trash2 className="w-4 h-4" />
                        </Button>
                      </div>
                    </div>

                    <p className="text-sm text-muted-foreground">
                      {review.comment}
                    </p>

                    {review.productId && (
                      <p className="text-xs text-muted-foreground mt-2">
                        Product ID: {review.productId}
                      </p>
                    )}
                  </div>
                </div>
              </Card>
            ))}
          </div>
        )}
      </Card>
    </div>
  );
}
