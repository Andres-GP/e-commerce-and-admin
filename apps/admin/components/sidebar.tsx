"use client"

import { LayoutDashboard, Package, ShoppingCart, Star, ShoppingBag } from "lucide-react"
import { cn } from "@/lib/utils"

interface SidebarProps {
  currentPage: string
  onPageChange: (page: "dashboard" | "products" | "orders" | "reviews" | "carts") => void
}

export function Sidebar({ currentPage, onPageChange }: SidebarProps) {
  const menuItems = [
    { id: "dashboard", label: "Dashboard", icon: LayoutDashboard },
    { id: "products", label: "Products", icon: Package },
    { id: "orders", label: "Orders", icon: ShoppingBag },
    { id: "reviews", label: "Reviews", icon: Star },
    { id: "carts", label: "Carts", icon: ShoppingCart },
  ]

  return (
    <aside className="w-64 bg-card border-r border-border flex flex-col animate-slide-in-left">
      <div className="p-6 border-b border-border">
        <h1 className="text-2xl font-bold bg-gradient-to-r from-primary to-chart-2 bg-clip-text text-transparent">
          Admin Panel
        </h1>
        <p className="text-sm text-muted-foreground mt-1">E-Commerce Management</p>
      </div>

      <nav className="flex-1 p-4 space-y-2">
        {menuItems.map((item) => {
          const Icon = item.icon
          const isActive = currentPage === item.id

          return (
            <button
              key={item.id}
              onClick={() => onPageChange(item.id as any)}
              className={cn(
                "w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200",
                "hover:bg-accent hover:scale-[1.02] active:scale-[0.98]",
                isActive && "bg-primary text-primary-foreground shadow-lg shadow-primary/20",
              )}
            >
              <Icon className={cn("w-5 h-5", isActive && "animate-pulse")} />
              <span className="font-medium">{item.label}</span>
            </button>
          )
        })}
      </nav>

      <div className="p-4 border-t border-border">
        <div className="flex items-center gap-3 px-4 py-3">
          <div className="w-10 h-10 rounded-full bg-gradient-to-br from-primary to-chart-2 flex items-center justify-center text-primary-foreground font-bold">
            A
          </div>
          <div>
            <p className="font-medium text-sm">Admin User</p>
            <p className="text-xs text-muted-foreground">admin@store.com</p>
          </div>
        </div>
      </div>
    </aside>
  )
}
