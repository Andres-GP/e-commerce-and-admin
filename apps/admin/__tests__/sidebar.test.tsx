import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Sidebar } from "@/components/sidebar";

describe("Sidebar Component", () => {
  const mockOnPageChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders the header and user info", () => {
    render(<Sidebar currentPage="dashboard" onPageChange={mockOnPageChange} />);

    expect(screen.getByText("Admin Panel")).toBeInTheDocument();
    expect(screen.getByText("E-Commerce Management")).toBeInTheDocument();
    expect(screen.getByText("Admin User")).toBeInTheDocument();
    expect(screen.getByText("admin@store.com")).toBeInTheDocument();
  });

  it("renders all menu items correctly", () => {
    render(<Sidebar currentPage="dashboard" onPageChange={mockOnPageChange} />);

    const menuItems = ["Dashboard", "Products", "Orders", "Reviews", "Carts"];
    menuItems.forEach((label) => {
      expect(screen.getByText(label)).toBeInTheDocument();
    });
  });

  it("highlights the active menu item", () => {
    render(<Sidebar currentPage="products" onPageChange={mockOnPageChange} />);

    const activeButton = screen.getByText("Products").closest("button");
    expect(activeButton).toHaveClass("bg-primary");
    expect(activeButton).toHaveClass("text-primary-foreground");
  });

  it("calls onPageChange when a menu item is clicked", () => {
    render(<Sidebar currentPage="dashboard" onPageChange={mockOnPageChange} />);

    const productsButton = screen.getByText("Products").closest("button");
    fireEvent.click(productsButton!);

    expect(mockOnPageChange).toHaveBeenCalledTimes(1);
    expect(mockOnPageChange).toHaveBeenCalledWith("products");
  });

  it("applies animation class to the active icon", () => {
    render(<Sidebar currentPage="reviews" onPageChange={mockOnPageChange} />);

    const activeIcon = screen.getByText("Reviews")
      .previousSibling as HTMLElement;
    expect(activeIcon).toHaveClass("animate-pulse");
  });

  it("does not apply animation class to inactive icons", () => {
    render(<Sidebar currentPage="orders" onPageChange={mockOnPageChange} />);

    const inactiveIcon = screen.getByText("Dashboard")
      .previousSibling as HTMLElement;
    expect(inactiveIcon).not.toHaveClass("animate-pulse");
  });

  it("renders the sidebar layout structure properly", () => {
    const { container } = render(
      <Sidebar currentPage="dashboard" onPageChange={mockOnPageChange} />
    );

    const aside = container.querySelector("aside");
    expect(aside).toHaveClass("flex", "flex-col");
    expect(aside).toBeInTheDocument();

    const nav = container.querySelector("nav");
    expect(nav).toBeInTheDocument();
    expect(nav?.querySelectorAll("button")).toHaveLength(5);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Sidebar currentPage="dashboard" onPageChange={mockOnPageChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
