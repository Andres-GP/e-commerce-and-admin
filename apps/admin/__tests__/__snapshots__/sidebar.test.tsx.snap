// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Sidebar Component matches snapshot 1`] = `
<DocumentFragment>
  <aside
    class="w-64 bg-card border-r border-border flex flex-col animate-slide-in-left"
  >
    <div
      class="p-6 border-b border-border"
    >
      <h1
        class="text-2xl font-bold bg-gradient-to-r from-primary to-chart-2 bg-clip-text text-transparent"
      >
        Admin Panel
      </h1>
      <p
        class="text-sm text-muted-foreground mt-1"
      >
        E-Commerce Management
      </p>
    </div>
    <nav
      class="flex-1 p-4 space-y-2"
    >
      <button
        class="w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200 hover:bg-accent hover:scale-[1.02] active:scale-[0.98] bg-primary text-primary-foreground shadow-lg shadow-primary/20"
      >
        <svg
          class="lucide lucide-layout-dashboard w-5 h-5 animate-pulse"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="9"
            rx="1"
            width="7"
            x="3"
            y="3"
          />
          <rect
            height="5"
            rx="1"
            width="7"
            x="14"
            y="3"
          />
          <rect
            height="9"
            rx="1"
            width="7"
            x="14"
            y="12"
          />
          <rect
            height="5"
            rx="1"
            width="7"
            x="3"
            y="16"
          />
        </svg>
        <span
          class="font-medium"
        >
          Dashboard
        </span>
      </button>
      <button
        class="w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200 hover:bg-accent hover:scale-[1.02] active:scale-[0.98]"
      >
        <svg
          class="lucide lucide-package w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 21.73a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73z"
          />
          <path
            d="M12 22V12"
          />
          <path
            d="m3.3 7 7.703 4.734a2 2 0 0 0 1.994 0L20.7 7"
          />
          <path
            d="m7.5 4.27 9 5.15"
          />
        </svg>
        <span
          class="font-medium"
        >
          Products
        </span>
      </button>
      <button
        class="w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200 hover:bg-accent hover:scale-[1.02] active:scale-[0.98]"
      >
        <svg
          class="lucide lucide-shopping-bag w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
          />
          <path
            d="M3 6h18"
          />
          <path
            d="M16 10a4 4 0 0 1-8 0"
          />
        </svg>
        <span
          class="font-medium"
        >
          Orders
        </span>
      </button>
      <button
        class="w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200 hover:bg-accent hover:scale-[1.02] active:scale-[0.98]"
      >
        <svg
          class="lucide lucide-star w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
          />
        </svg>
        <span
          class="font-medium"
        >
          Reviews
        </span>
      </button>
      <button
        class="w-full flex items-center gap-3 px-4 py-3 rounded-lg transition-all duration-200 hover:bg-accent hover:scale-[1.02] active:scale-[0.98]"
      >
        <svg
          class="lucide lucide-shopping-cart w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="21"
            r="1"
          />
          <circle
            cx="19"
            cy="21"
            r="1"
          />
          <path
            d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
          />
        </svg>
        <span
          class="font-medium"
        >
          Carts
        </span>
      </button>
    </nav>
    <div
      class="p-4 border-t border-border"
    >
      <div
        class="flex items-center gap-3 px-4 py-3"
      >
        <div
          class="w-10 h-10 rounded-full bg-gradient-to-br from-primary to-chart-2 flex items-center justify-center text-primary-foreground font-bold"
        >
          A
        </div>
        <div>
          <p
            class="font-medium text-sm"
          >
            Admin User
          </p>
          <p
            class="text-xs text-muted-foreground"
          >
            admin@store.com
          </p>
        </div>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;
